idioms/str-vs-string.md

```diff
@@ -0,0 +1,89 @@
+# Using &str instead of &String
+
+## Description
+
+It is usually recommended to prefer a `&str` type over a `&String` type for arguments.  In most cases `&str` has more flexibility.  An `&String` will coerce to a `&str` type when required -- however, the oppositive is not true.  
```

Comment:
From here on out, I will claim that auto-correct is responsible for every typo I make :) thanks!